frontend/src/metabase/dashboard/components/PublicLinkPopover/PublicLinkPopover.tsx

```diff
@@ -52,70 +46,24 @@ export const PublicLinkPopover = ({
   };
 
   return (
-    <Popover opened={isOpen} onClose={onClose}>
+    <Popover opened={isOpen} onClose={onClose} position="bottom-end">
       <Popover.Target>
         <div>{target}</div>
       </Popover.Target>
       <Popover.Dropdown>
-        <Box p="lg" data-testid="public-link-popover-content">
+        <Box p="lg" w="28rem" data-testid="public-link-popover-content">
           <Title order={4}>{t`Public link`}</Title>
           <Text>{t`Anyone can view this if you give them the link.`}</Text>
-          <LinkContainer noWrap w="28rem" p="sm" align="center">
-            {loading ? (
-              <PublicLinkTextContainer>
-                <Text truncate c="text.0">{t`Loading…`}</Text>
-              </PublicLinkTextContainer>
-            ) : (
-              <>
-                <PublicLinkTextContainer>
-                  <Text truncate data-testid="public-link-text">
-                    {url}
-                  </Text>
-                </PublicLinkTextContainer>
-                <PublicLinkCopyButton value={url} />
-              </>
-            )}
-          </LinkContainer>
-          {extensions && extensions.length > 0 && (
-            <Group my="sm">
-              {extensions.map(extension => (
-                <ExtensionOption
-                  data-testid="extension-option"
-                  key={extension}
-                  isSelected={extension === selectedExtension}
-                  onClick={() =>
-                    setSelectedExtension?.(
-                      extension === selectedExtension ? null : extension,
-                    )
-                  }
-                >
-                  {extension}
-                </ExtensionOption>
-              ))}
-            </Group>
-          )}
-          {isAdmin && (
-            <Box mt="md">
-              <Tooltip
-                py="0.6rem"
-                px="0.75rem"
-                bg="bg.3"
-                withArrow
-                arrowSize={10}
-                label={
-                  <Text
-                    fw={700}
-                    fz="md"
-                    color="white"
-                  >{t`Affects both public link and embed URL for this dashboard`}</Text>
-                }
-              >
-                <Anchor fz="sm" c="error.0" fw={700} onClick={onRemoveLink}>
-                  {t`Remove this public link`}
-                </Anchor>
-              </Tooltip>
-            </Box>
-          )}
+          <PublicLinkCopyPanel
+            loading={loading}
+            url={url}
+            onRemoveLink={isAdmin ? onRemoveLink : undefined}
+            extensions={extensions}
+            selectedExtension={selectedExtension}
+            onChangeExtension={setSelectedExtension}
+            removeButtonLabel={t`Remove public link`}
+            removeTooltipLabel={t`Affects both public link and embed URL for this dashboard`}
```

Comment:
Do you think these labels would be dynamic? Or put it another way, would `PublicLinkCopyPanel` be used somewhere else that will have different labels? Otherwise, I think hard coding them in `PublicLinkCopyPanel` for simplicity would make more sense. What do you think?

Reply:
I guess the only issue here is that we don't have the resource available to tell which text we should provide - when we were just passing the dashboard/question it would have been a lot easier, but with just passing the URL that becomes trickier. We could check the substring for "question" or "dashboard" but I think that will just make things more complicated than they need to be

Reply:
yeah good point, I missed that. because currently they look like hard-coded strings.